### PR TITLE
PY3 changes

### PR DESCRIPTION
**Pull request by nyov**

### CODE REVIEW

File: tests/requirements-py3.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pytest>=2.6.0
+pytest-twisted
+testfixtures
+jmespath
```
Comment (kmike) [inline tests/requirements-py3.txt]: Why is jmespath a test requirement, shouldn't it be a regular requirment? 
Sorry if you've answered me already and I forgot the answer :)

Comment (nyov) [inline tests/requirements-py3.txt]: Sorry, I thought the question wasn't pointed at me.
I don't have the answer -- I just followed the same order as requirements.txt,
it's in the `tests/requirements.txt` file for python2, too.

```
$ git grep jmespath
docs/topics/loaders.rst:    Requires jmespath (https://github.com/jmespath/jmespath.py) to run.
scrapy/loader/processors.py:        Query the input string for the jmespath (given at instantiation),
scrapy/loader/processors.py:        Requires : jmespath(https://github.com/jmespath/jmespath)
scrapy/loader/processors.py:        import jmespath
scrapy/loader/processors.py:        self.compiled_path = jmespath.compile(self.json_path)
scrapy/loader/processors.py:        """Query value for the jmespath query and return answer
scrapy/loader/processors.py:        :return: Element extracted according to jmespath query
tests/requirements-py3.txt:jmespath
tests/requirements.txt:jmespath
```
Seeing as it's not really important (only used in some itemloader proc), I'd prefer to not pull it in as a global installation requirement.

**Comment by kmike** [comment]
Hey @nyov! 

* [x] *split requirements into files* - looks good;
* [ ] *fix test cmdline* - changes look good. It seems we don't have tests for `--profile` (see `_run_command_profiled` and `scrapy.xlib.lsprofcalltree`). To port cmdline we should add a test for this option and make sure it works.
* [ ] *renames (six types)* - just some renames using six library - it looks good, but it seems there is a conflict with @dangra's merged cookies PR;
* [x] *fix test loader* - LGTM;
* [ ] *fix tests pipelines files* - Referrer header should be decoded for nicer log messages. There is also a tricky part when headers are passed to boto, but it looks like we're fine here. I also wonder if we should use canonicalize_url to create a hash (as we do in fingerprints), but this is out of scope here.
* [x] *fix tests pipelines images* - looks good, depends on files pipeline;
* [x] *response bodies as bytes* - looks good.

**Comment by kmike** [comment]
@nyov I can either cherry-pick some of the commits or wait - what do you prefer?

**Comment by dangra** [comment]
> @nyov I can either cherry-pick some of the commits or wait - what do you prefer?
>  Show all checks

+1 to cherry pick :)

**Comment by nyov** [comment]
+1 for cherry-pick here too. Sorry (for the work on your part).
I don't need the attribution.

**Comment by kmike** [comment]
Merged in #1415. Thanks @nyov!

**Pull request closed without merging**